fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaSendFragment.java

```diff
@@ -60,12 +63,26 @@ private void sendNewChirp() {
     // will
     // make a toast appear and it will log the error
     if (mSocialMediaViewModel.getLaoId().getValue() == null) {
-      ErrorUtils.logAndShow(getContext(), TAG, R.string.error_no_lao);
+      ErrorUtils.logAndShow(requireContext(), TAG, R.string.error_no_lao);
     } else {
-      mSocialMediaViewModel.sendChirp(
-          mSocialMediaSendFragBinding.entryBoxChirp.getText().toString(),
-          null,
-          Instant.now().getEpochSecond());
+      mSocialMediaViewModel.addDisposable(
+          mSocialMediaViewModel
+              .sendChirp(
+                  mSocialMediaSendFragBinding.entryBoxChirp.getText().toString(),
+                  null,
+                  Instant.now().getEpochSecond())
+              .subscribe(
+                  msg -> {},
+                  error -> {
+                    if (error instanceof KeyException
+                        || error instanceof GeneralSecurityException) {
+                      ErrorUtils.logAndShow(
+                          requireContext(), TAG, error, R.string.error_sending_chirp);
+                    } else {
+                      ErrorUtils.logAndShow(
+                          requireContext(), TAG, error, R.string.error_post_transaction);
+                    }
```

Comment:
I think it's not immediately clear from the exception names why these conditional statements are so

Reply:
I think I just used the wrong error message